test/Algebra/Graph/Test/Acyclic/Labelled/AdjacencyMap.hs

```diff
@@ -0,0 +1,237 @@
+{-# LANGUAGE ViewPatterns #-}
+-------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Test.Acyclic.Labelled.AdjacencyMap
+-- Stability  : experimental
+--
+-- Testsuite for "Algebra.Graph.Acyclic.Labelled.AdjacencyMap".
+-------------------------------------------------------------
+
+module Algebra.Graph.Test.Acyclic.Labelled.AdjacencyMap (
+  testAcyclicLabelledAdjacencyMap 
+  ) where
+    
+import Data.Monoid
+
+import Algebra.Graph.Acyclic.Labelled.AdjacencyMap
+import Algebra.Graph.Test
+import Algebra.Graph.Label
+
+import qualified Data.List as List
+import qualified Data.Set as Set
+
+type S = Sum Int
+type D = Distance Int
+
+type LAS = AdjacencyMap S   Int
+type LAD = AdjacencyMap D   Int
+
+-- TODO: Switch to using generic tests.
```

Comment:
I could start with this one but I'm not sure what you mean. Is it about adding it to `API.hs`?
Also, how should I incorporate my change to your branch?

Reply:
> I could start with this one but I'm not sure what you mean. Is it about adding it to API.hs?

I suggest you leave that one as well. One needs to make separate generic tests cases for acyclic graphs as the acyclic API does not have edges yet (or) make a `*.Ord` module similar to `Acyclic.Ord` to make a complete API using functions from both the modules. Creating an issue regarding this would help start a discussion.

I would also suggest you leave out `TODO: Make 'skeleton :: Ord a => AdjacencyMap e a -> AAM.AdjacencyMap a'` and `TODO: Add 'fromGraph'`. In light of the recent PR#223, some functions might be removed.

`TODO: Add additional/change examples.` This would be a good place to start. This PR needs more examples and the corresponding tests to be written.

> Also, how should I incorporate my change to your branch?

If it is not too much hassel, one way to do this is to create a PR to my fork of the repository. I'll merge the PR accordingly.

Please let me know if you have any other method in mind.

Reply:
@Avasil It's a bit difficult to contribute to this part of the codebase at the moment because both @adithyaov and I are making some changes. I can try to find an interesting isolated task for you if you like :)

Reply:
That makes sense, I'll wait a bit until it stabilizes.

> I can try to find an interesting isolated task for you if you like :)

@snowleopard That would be perfect!

Reply:
@Avasil Here is an interesting issue which I haven't had time to think about: #141. It would be nice to make an experiment about whether it's practical to implement a generic `AdjacencyMap a` by keeping `AdjacencyIntMap` internally together with a bijection between `a` and `Int`.